sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala

```diff
@@ -154,8 +151,163 @@ object GpuScalar {
       case t => throw new IllegalStateException(s"$t is doesn't support NaNs")
     }
   }
+
+  /**
+   * The following 2 APIs are used to create a GpuScalar from a Scala `Any`.
```

Comment:
Done. I removed this `converted`. Since it will throw exception if the data can not be supported.
Spark Literal has a one-to-one mapping from the value to its type according to the function [validateLiteralValue](https://github.com/apache/spark/blob/branch-3.0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala#L178), which can simplify the data interpretation a lot for each type. I just wanted to follow this but seems it is not needed. Because the `GpuScalar.from`  has its own mapping.
)